### ISSUE TITLE
seems this new update is very simple for instagram to detect

### ISSUE THREAD

**Issue by yahgiggle** [instabot-py/instabot.py#2327]
hey guys this is not a bug as such, its just to talk about the fact that Instagram is detecting this latest version, after about 3 likes its blocked and i get banned, even with the setting at 50 likes per day with no comments or follows, it seems to detect it somehow any ideas how ?  seems it does not matter if you change ip address or computer or anything even if you are unblocked then delete the session file before you run the bot it then bans you after about 3 likes, ive notice that when this happens there is no cool down anymore and the bot goes into like crazy mode where it will keep liking every sec, what ever happened to the cool down when a like does not work, i wonder if this is whats causing the problem

**Comment by AndrewJacob99** [comment]
Unfortunately the same problem here :(

**Comment by yahgiggle** [comment]
I'm going to go back to an older version one that never used the session
file idea and had the cool down I used that for months with out a problem oO

**Comment by svinc** [comment]
@yahgiggle I've been running on the old version, Instagram detects it too :(

**Comment by nativecodeabuser** [comment]
I suggested it in another post. Bots are performing actions in even, predictable intervals. Server-side it must be so obvious it takes 3 actions to detect...

**Comment by AndrewJacob99** [comment]
I tried to delete "username".session file and start the bot over as suggested in previous issues but now Instagram doesn't recognize my device and after using the verification code they sent me it says wrong code.

**Comment by svinc** [comment]
@AndrewJacob99 Deleting your session file/changing your IP will no longer help you. IG outsmarted that.

**Comment by svinc** [comment]
@nativecodeabuser That's a smart idea. I've noticed a pattern for my accounts, they run for 10-12 hours and then IG detects them. IG must be timing how often they perform actions.

**Comment by creestee** [comment]
how´s it going? Does it still get blocked after the first likes?

**Comment by nativecodeabuser** [comment]
I'm currently about 500 likes in the process and still going fine. Randomizing with a very dirty solution in bash (starting and killing bot process). Would be nice to have that built in.

**Comment by svinc** [comment]
@nativecodeabuser Could you forward me a pastebin of your code? And as of right now, yes it does. I'm currently debugging by connecting to the same proxy that the bot connects to. When I connect from my regular browser it works fine, when the bot connects it doesn't. Clearly the bot isn't sending something that the browser is.

**Comment by yahgiggle** [comment]
I made a small bat file that closes the bot at random but didn't work
Instagram still detected it only takes 3 likes to detect then it blocks the
likes and sees the bot trying to like every second then bans me

**Comment by svinc** [comment]
@yahgiggle I'm gonna be double checking the headers the bot sends and how it interacts with IG's APi. I'll let you know when I code up a solution but I have a pretty good idea of why IG is blocking the bot.

**Comment by creestee** [comment]
i hope there will be an update for the bot with an randomizer. But maybe IG is watching out for something different...

**Comment by yahgiggle** [comment]
Good stuff

**Comment by svinc** [comment]
> i hope there will be an update for the bot with an randomizer. But maybe IG is watching out for something different...

The bot already has a randomizer built in, I've ruled it out. That's not the issue.

**Comment by yahgiggle** [comment]
Yeah my bat file made it even more random so definitely something else
going on,  I had to edit the database for the old version I changed the
browser info as Instagram keeped suspecting something was up, this new
version I noticed does not use the DB to store the fake browser info so I
wonder if this is how it's getting detected

**Comment by svinc** [comment]
@yahgiggle I don't think thats how, my suspicion is that it's a missing header. Testing now, will keep you updated.

**Comment by svinc** [comment]
Yup, from my observations it's missing the "x-ig-app-id" header. I've manually added my own into the bot and it started working. It needs to auto generate it though, gonna have to figure out how to do that.

**Comment by yahgiggle** [comment]
Cheers, looking forward to it, and I really appreciate it, dam I should
learn python I know Lua Java HTML PHP so learning another one wouldn't hurt
lol 😂 I'm just stuck atm and can't be bothered oO

**Comment by yahgiggle** [comment]
Good stuff

**Comment by creestee** [comment]
@svinc please let us know if you got more

**Comment by AtlasCoelestis** [comment]
Here the same problem only with one of my 3 pages.

**Comment by blauergolfer** [comment]
I also face this issue...
Can one of you please try following:

- Login to instagram on your desktop browser
- Open a picture
- Open the developer tools
- Like the image

The heart turns red but in I get a 400 response with the same message that also is shown in the bot logs...
So maybe all desktop interactions are blocked, once you have been caught...
What do you think?

**Comment by creestee** [comment]
@blauergolfer for me it works just fine on the website

**Comment by bboy519** [comment]
found a way to fix this. Maybe temporarily
1. Open browser, login Instagram and switch UA to "Instagram 10.19.0 (iPhone9,4; iOS 10_3_1; en_SI; en-SI; scale=2.88; gamut=wide; 1080x1920) AppleWebKit/420+" and refresh. Instagram may log you out but don't worry, try to log in again. You may face a login loop and just ignore it.
2. Switch User Agent to your regular UA and like 400 will disappear.

**Comment by bboy519** [comment]
UPDATE: There is a way to spoof UA as IG built in browser. Successfully logged in.
Put "https://wtools.io/check-my-user-agent" to the "website"  blank
then copy the UA and apply it to the bot
That's it :)

**Comment by bboy519** [comment]
UPDATE: There is a way to spoof UA as IG built in browser. Successfully logged in.
Put "https://wtools.io/check-my-user-agent" to the "website"  blank;
Open the link in IG;
then copy the UA and apply it to the bot;
That's it :)
** It worked with the old version (python 2.7) bot by modifying ./src/userinfo.py by adding user_agent = ("the_ua_shown_above" "")

--------------------------------------------------------

01.08.2019_10:40 Removing already liked medias..
01.08.2019_10:40 Trying to like media: 2100927239763767097
01.08.2019_10:40 Liked: 2100927239763767097. Like #1.
01.08.2019_10:42 Trying to like media: 2100926144195700601
01.08.2019_10:42 Liked: 2100926144195700601. Like #2.
01.08.2019_10:43 Trying to like media: 2100916801098352991
01.08.2019_10:43 Liked: 2100916801098352991. Like #3.

----------------------------------------------------------
Request URL: https://www.instagram.com/web/likes/the_post_code/like/?hl=en
Request Method: POST
Status Code: 200

**Comment by mrallanodreman** [comment]
@bboy519 how can i do this on instagram?

**Comment by bboy519** [comment]
@mrallanodreman  As I figured out the blockage l'll make a video on YouTube

**Comment by bboy519** [comment]
@mrallanodreman Here it is
https://youtu.be/qnBg4aiCVdA

**Comment by bboy519** [comment]
Just firgured out a way to fix instagram 400 like error.

Right click - Inspect - Network - Network Conditions - User Agent
Uncheck "select automatically"
setect "custom"
Enter the following string:

Instagram 10.19.0 (iPhone9,4; iOS 10_3_1; en_SI; en-SI; scale=2.88; gamut=wide; 1080x1920) AppleWebKit/420+

Then refresh and try to re-login a couple of times
After that, check the "select automatically" back and login your account. 
That's it!

I've made a video tutorial here:
https://youtu.be/qnBg4aiCVdA

**Comment by mrallanodreman** [comment]
@bboy519 im on android dude!!!!

**Comment by bboy519** [comment]
@mrallanodreman It seems UA spoofing isn't working.
IG detected "Chrome OS · Chrome " and
01.08.2019_11:58 Login error! Check your login data!
01.08.2019_11:58 Removing already liked medias..
01.08.2019_11:58 Removing already liked medias..

**Comment by AtlasCoelestis** [comment]
I notice that if i delete my "user.session" before start the instabot likes are working again

**Comment by bboy519** [comment]
> I notice that if i delete my old "user.session" before start the instabot likes are working again.

But soon the bot will stop working

**Comment by AtlasCoelestis** [comment]
> > I notice that if i delete my old "user.session" before start the instabot likes are working again.
> 
> But soon the bot will stop working. It's like that IG has upgraded their bot detection module.

Yes, but only on my personal profile, i have other 2 instagram pages and here work all very fine, i don't know why only 1 of 3 have this kind of problems.
And don't know a long term solution for this unfortunately..

**Comment by svinc** [comment]
@bboy519 Spoofing UA is not a permanent solution.

**Comment by svinc** [comment]
Also to clarify, IG has increased their bot detection significantly. My account has been blocked for the next 7 days and it displays a new error message stating something along the lines of "Your account violated community guidelines for using a service that helps you get more likes/followers". I'm gonna have to run the bot on test accounts.

**Comment by svinc** [comment]
I'm gonna be testing a host of different solutions. My theory is IG knows that no person can send requests 24/7, the bot needs to act like a human being to an extent and act in bursts. So it'll get the same amount of actions out just at different time intervals than before.

**Comment by guztavoalves** [comment]
I deleted my session file and not get login anymore =/

`
2019-08-01 16:13:29,710 - InstaBot - INFO - Checking for updates...
2019-08-01 16:13:29,754 - InstaBot - INFO - You are running the latest stable version
2019-08-01 16:13:29,754 - InstaBot - INFO - Instabot v1.2.6/0 started at 01.08.2019 16:13:
2019-08-01 16:13:29,754 - InstaBot - INFO - Trying to login as foo ...
2019-08-01 16:13:30,826 - InstaBot - ERROR - Something is wrong with Instagram! Please try again later...
2019-08-01 16:13:30,826 - InstaBot - ERROR - Error =>Sorry, there was a problem with your request.
2019-08-01 16:13:34,148 - InstaBot - INFO - Exit Program... GoodBye
`

I trying login with different proxy and spoofing UA, but nothing.

**Comment by guztavoalves** [comment]
I deleted my session file and not get login anymore =/

2019-08-01 16:13:29,710 - InstaBot - INFO - Checking for updates... 
2019-08-01 16:13:29,754 - InstaBot - INFO - You are running the latest stable version 
2019-08-01 16:13:29,754 - InstaBot - INFO - Instabot v1.2.6/0 started at 01.08.2019 16:13: 
2019-08-01 16:13:29,754 - InstaBot - INFO - Trying to login as foo ... 
2019-08-01 16:13:30,826 - InstaBot - ERROR - Something is wrong with Instagram! Please try again later... 
2019-08-01 16:13:30,826 - InstaBot - ERROR - Error =>Sorry, there was a problem with your request. 
2019-08-01 16:13:34,148 - InstaBot - INFO - Exit Program... GoodBye

I trying login with different proxy and spoofing UA, but nothing.

**Comment by creestee** [comment]
So on my PC it does not work but I just tried it in Termux on my phone and it seems to work perfectly fine there...

**Comment by mrallanodreman** [comment]
On termux android is working perfectly for accounts that not are banned before... 

My personal acc still block from termux... im triying deleting de session.file and im getting this error 

______Starting bot_____
2019-08-01 11:35:55,897 - InstaBot - INFO - Checking for updates...
2019-08-01 11:35:57,836 - InstaBot - INFO - You are running the latest stable version
2019-08-01 11:35:57,837 - InstaBot - INFO - Instabot v1.2.6/0 started at 01.08.2019 11:35:
2019-08-01 11:35:57,842 - InstaBot - INFO - Trying to login as mr.polemic...
2019-08-01 11:36:04,021 - InstaBot - ERROR - Something is wrong with Instagram! Please try again later...
2019-08-01 11:36:04,022 - InstaBot - ERROR - Error =>Sorry, there was a problem with your request.
2019-08-01 11:36:04,026 - InstaBot - INFO - Exit Program... GoodBye
$

**Comment by svinc** [comment]
@creestee @mrallanodreman that's a temporary solution. If you run it for 5-6 hours you will get banned. Do not do that.

**Comment by mrallanodreman** [comment]
Im always run the bot on my android terminal, i got less problems for some reazon, in this case only my personal acc have a bot-ban  and i think is for the several actions

**Comment by mrallanodreman** [comment]
Update: 

If you delete the session.file you are lost!

The login is imposible!!!

**Comment by yahgiggle** [comment]
for anyone wanting the code for random stops and starts i run a app called
restart on crash - this will restart your startbot.bat file if it sees the
file as been closed this is a way to loop your random code with very little
coding needed

 title startupbots
COLOR 27
set /a A=%random%%% 3600
TIMEOUT %A%
echo Bot Started
start C:\Instagram-bot\Models\Start.bat
set /a B=%random%%% 30 +30
TIMEOUT %B%
start C:\Instagram-bot\Photos\Start.bat
set /a C=%random%%% 14400
TIMEOUT %C%
echo end bot
start C:\Windows\System32\taskkill.exe /t /f /im cmd.exe


i run two bots ^^^ each bot has a start bat file this just adds color to
the cmd and names it so i can tell what bot is what

the code is simple for the start.bat

the code for this is

title Photos
COLOR 37
C:\Instagram-bot\Photos\example.py

**Comment by svinc** [comment]
I've confirmed that my solution has been working for the past 6 hours. Will see how it holds up in the morning. So far really promising results!

**Comment by yahgiggle** [comment]
Only one of my accounts is blocked so it seems Instagram is not detecting
it on all accounts my guess is they use a bot that cracks down on accounts
that have over a set number of followers

**Comment by bboy519** [comment]
Noticed that if login get blocked the only way to get around it is to change the IP address. This is extreme difficult when hosting on a VPS since cloud providers usually does't allow IP change.

**Comment by yahgiggle** [comment]
second account is also now blocked did not even like 30 images over a 24
hours and it got blocked too

**Comment by blauergolfer** [comment]
The problem is that everything we write here can also be seen by IG.
And if I were a IG dev. responsable for bot detection, github would be the first place to search for....

Think about it :-)

**Comment by yahgiggle** [comment]
Lol 😂 that's not how it works code is code they don't have the man power
to worry about this site it's much better that they use AI to detect bot
activities we just need to always stay one steep ahead but that gets harder
and harder to do

**Comment by svinc** [comment]
It’s been 17 hours, everything still running smoothly. Instagram hasn’t detected my solution. I’ll keep you guys updated.

**Comment by svinc** [comment]
@blauergolfer it’s impossible for IG to 100% truly detect bots. You just have to be one step ahead.

**Comment by bboy519** [comment]
Using an old version of the bot seems working (First login). But if logged out I won't be able to login again unless IP change is possible.

----------------------------------------------

02.08.2019_04:19 Trying to like media: 2101444211993142077
02.08.2019_04:19 Liked: 2101444211993142077. Like #94.
02.08.2019_04:20 Trying to like media: 2101443078021142460
02.08.2019_04:20 Liked: 2101443078021142460. Like #95.
02.08.2019_04:21 Trying to like media: 2101442155651508615
02.08.2019_04:21 Liked: 2101442155651508615. Like #96.
02.08.2019_04:22 Trying to like media: 2101440818026920388
02.08.2019_04:22 Liked: 2101440818026920388. Like #97.
02.08.2019_04:24 Trying to like media: 2101441551830752758
02.08.2019_04:24 Liked: 2101441551830752758. Like #98.
02.08.2019_04:25 Trying to like media: 2101440935576904914
02.08.2019_04:25 Liked: 2101440935576904914. Like #99.

**Comment by etaxi341** [comment]
I am sure the Method of @svinc does not work. I am trying to fix this on my own program since 2 weeks... My program is not a bot. So it is not autoliking whatsoever. Still my accounts get feedback_required after beeing logged in more than 48 hours...

**Comment by svinc** [comment]
@etaxi341 We’ll have to wait and see. If my solution works for more than 48 hours is it solid by your standards?

**Comment by svinc** [comment]
@bboy519 how long was that working for?

**Comment by bboy519** [comment]
@svinc currently 4 hours. I'll keep observing if there's any suspected errors.

**Comment by etaxi341** [comment]
> @etaxi341 We’ll have to wait and see. If my solution works for more than 48 hours is it solid by your standards?

Not sure... But if your method really works. It's weird that my program does not. Even if the account is afk for 48hours and then tries to like an image or upload something it does not work. But I will keep on looking if you have any news about your method :)

**Comment by AtlasCoelestis** [comment]
Last night bot working for 7h without any problems. From 18:00pm to 01:00am.

**Comment by creestee** [comment]
Can someone please explain to me why deleting the .session file isn´t an option. If I delete it before every restart it works.

**Comment by creestee** [comment]
Also it´s possible that IG is testing bot detection for only some accounts like they do it with design

**Comment by bboy519** [comment]
Instagram blacklisted my account. No way to log in  new clients. Showing “theres a problem with your account”. Not even giving email codes.

I've reactivated my old account and it logged in without problem, after imputing the security code.

**Comment by husligc** [comment]
> It’s been 17 hours, everything still running smoothly. Instagram hasn’t detected my solution. I’ll keep you guys updated.

let us know man haha!

**Comment by rrdd** [comment]
> Yup, from my observations it's missing the "x-ig-app-id" header. I've manually added my own into the bot and it started working. It needs to auto generate it though, gonna have to figure out how to do that.

Is this still working? I added this header, but was blocked very fast. Where did you add the x-ig-app-id header?

**Comment by neospeed83** [comment]
@svinc I hope your bot fix works. Hoping 48 hrs go fine 😅

**Comment by yurilaaziz** [comment]
I release new package of instabot-py, please confirm using the new code

**Comment by mrallanodreman** [comment]
Confirm what? What changes are made? Im working fine since yesterday... only likes!!!

**Comment by mrallanodreman** [comment]
Update: down again

**Comment by Pencilhead12** [comment]
yeah @svinc let us know if your method is successful, I've turned off my bot for the past few days and I miss it :/

**Comment by elmissouri16** [comment]
I dont know much about how this work but on the header they are 2 values  'X-Instagram-AJAX' and 
    'X-IG-App-ID' the first one we send with value of 1 and from monitoring http requests of instagram on devtool its always generate unique id and for X-IG-App-ID thats have unique value i test it on chrome and firefox same ip its sent always same value i notice the library for  X-Instagram-AJAX always sent withvalue 1 and X-IG-App-ID not been used at all can be this the problem  ?  @yurilaaziz  @svinc   and sorry for the language im not thats good on english

**Comment by yurilaaziz** [comment]
I fixed the main issue and I added some random wait between actions.
Please update using this article 
https://medium.com/@asker_amine/whats-new-in-the-instagram-bot-0-5-2-fe52154fdc8a

**Comment by husligc** [comment]
I think Instagram just made a new update.  I was getting not liked 400 and then received the random message to change my password (we think you've been using an external service etc etc).  

I changed my password and logged in using instabot again...

Received a new message on Instagram (your account has been compromised, change password or we will do x y and z). I've never seen this message before so i am probably going to lay off for awhile.  I think we need a whole new instabot update to combat this.  What do you guys think?

**Comment by creestee** [comment]
for me deleting the session file every time works best but that can´t work forever...

**Comment by Pencilhead12** [comment]
Have you all managed to get the bot to work again with it's most recent update? I run the bot through the examply.py file, which allows me to blacklist certain users, accurately choose a number of likes and comments per day, provide a list of hashtags from which to find posts, and generate comments from a choice of my own words etc. The new update of instabot.py has rendered the example.py redundant, do you have any advice on what I can do to operate as I once did? Cheers 🙏

**Comment by svinc** [comment]
> I fixed the main issue and I added some random wait between actions.
> Please update using this article
> https://medium.com/@asker_amine/whats-new-in-the-instagram-bot-0-5-2-fe52154fdc8a

What was the main issue?

**Comment by svinc** [comment]
> yeah @svinc let us know if your method is successful, I've turned off my bot for the past few days and I miss it :/

It's successful.

1. Make sure you pass the X-IG-APP-ID header.
2. The comment generation system is flawed, IG is able to detect it.
3. Also, modify the timing of the bot to make it more human-like.

There are other things I'd recommend such as an automatic rate limit system where it updates the limits automatically instead of leaving it up to the end-user.

**Comment by prodigeris** [comment]
@svinc 

> Make sure you pass the X-IG-APP-ID header.
> Also, modify the timing of the bot to make it more human-like.

Can we do that without adjusting the source code?

**Comment by svinc** [comment]
@prodigeris no, you're forced to.

**Comment by elmissouri16** [comment]
@svinc did u test the header with   X-IG-APP-ID ??

**Comment by husligc** [comment]
> > yeah @svinc let us know if your method is successful, I've turned off my bot for the past few days and I miss it :/
> 
> It's successful.
> 
> 1. Make sure you pass the X-IG-APP-ID header.
> 2. The comment generation system is flawed, IG is able to detect it.
> 3. Also, modify the timing of the bot to make it more human-like.
> 4. User-agent shouldn't be regenerated on each bot login, that's way too suspicious. It should be kept throughout sessions.
> 
> There are other things I'd recommend such as an automatic rate limit system where it updates the limits automatically instead of leaving it up to the end-user. I'm also unsure about whether or not the newest version of the bot introduces any new issues so there may be more things to change. This is just what worked for me on a really old version. I no longer update my bot because I've built a lot of things on top of it and it would be too time-consuming to re-modify.

Can we get a copy of yours?

**Comment by creestee** [comment]
I changed the user agent to the one my phone has and it works

**Comment by svinc** [comment]
@husligc It has just failed. When taking a closer look, I noticed the cookies the bot has are completely opposite of what's necessary. 
It's missing the following cookies:
ds_user_id
mid
rur
sessionid
urlgen
ig_cb (this should be set to 1), means you accepted IG's cookie policy

if someone can do research and figure out how to generate each one lmk.

**Comment by svinc** [comment]
To clarify, the issue with my solution occurs when i relogin. It can work fine for days but if I shut off the bot and try to start again, instant 400. For this reason, I suspect it’s something with either headers or cookies.

**Comment by elmissouri16** [comment]
X-Instagram-AJAX only have generated value the first time when they are no session file generated when u login with session file its well set to 1

**Comment by mrallanodreman** [comment]
@creestee hiw you did that¡
Are you on telegram group?

**Comment by mrallanodreman** [comment]
@svinc wich version are you running?

**Comment by svinc** [comment]
@elmissouri16 aha! could that be the issue?

**Comment by svinc** [comment]
@mrallanodreman a very old one from February

**Comment by husligc** [comment]
@svinc alright that is unfortunate! Let me know if you can get it to work.  Most of this exceeds my ability haha.  Received a week like/comment/post ban on one of my accounts, so I'm waiting.

**Comment by elmissouri16** [comment]
@svinc  maybe same as  u I'm using an old version of the API I'm trying to find the best approach to save the value of X-Instagram-AJAX and use when login with session file

**Comment by svinc** [comment]
@elmissouri16 I’m doing research on it. It’s generated on login (auth_switcher) and it’s the “rollout_hash”. I’ll keep you updated with my progress. But personally everything works for me except when i re-login which I have to do. If I constantly stay logged in that looks suspicious to IG

**Comment by elmissouri16** [comment]
and for the X-IG-App-ID i still not sure is it fixed value or generated i login from different IPs and different accounts ad was still same value did u have an idea about it @svinc

**Comment by elmissouri16** [comment]
> 
> 
> @elmissouri16 I’m doing research on it. It’s generated on login (auth_switcher) and it’s the “rollout_hash”. I’ll keep you updated with my progress. But personally everything works for me except when i re-login which I have to do. If I constantly stay logged in that looks suspicious to IG

same if find something new i well posted here

**Comment by svinc** [comment]
@elmissouri16 X-ig-app-ID is static. You can keep the same value across all accounts.

**Comment by elmissouri16** [comment]
> 
> 
> @elmissouri16 X-ig-app-ID is static. You can keep the same value across all accounts.

ok got it thats whats i did

**Comment by TriptonPt** [comment]
Has any of you guys considered ig bot detection is now using mouse movement (or lack of in this case) to raise flags? I was using a chrome extension for a few weeks and it was working perfectly fine but now it doesnt work no matter what, blocks everytime. Changing browser or computer doesnt seem to make any difference. I think they updated the bot detection even further and I doubt any bot not using a solution like selenium to emulate mouse behavior will work .

**Comment by svinc** [comment]
@TriptonPt they don’t do that. Ive checked + it’s too resource intensive to be practical for them to do

**Comment by prodigeris** [comment]
@TriptonPt It's an API. How and why would they do that?
I would suggest you download [Charles](https://www.charlesproxy.com/) and sniff out the traffic and see discrepancies between real app and bot.

**Comment by Pencilhead12** [comment]
When someone creates a working version, please share the code with us... I am unexperienced in code and have no idea what you are all talking about 😂 so I can't edit it by myself

**Comment by creestee** [comment]
After I changed the user agent it even works after restarting and without deleting the .session file!

To change it you have to edit instabot.py:
Search for `self.user_agent =` and replace `random.sample(config.get("list_of_ua"), 1)[0]` with your devices ua (you can look it up in the internet).

**Comment by TriptonPt** [comment]
@svinc Idk, it sure looks like it . How does even an imacro running exactly the same follow commands I do manually get blocked and detected so easily ? The only thing I see missing is random mouse movement and clicks . I got blocked within 3 follows with an imacro just pressing follow button on a user list I have in a text file and then after logging out I repeated the same process manually and did 20 follows without issues .

**Comment by svinc** [comment]
@TriptonPt Disable comments.

My theory is, IG is detecting repetitive comments. Try it out and let me know!

**Comment by svinc** [comment]
@prodigeris IG could do it, google recaptcha already does it.

**Comment by TriptonPt** [comment]
@svinc not using any comments I only do follow, pretty sure IG is detecting mouse actions . Just tried with a macro software and so far looks promising . I believe it could also be done with selenium but for it to work it would have to emulate real mouse movement and clicks on screen, with randomization of the click area within the button margins .

Technology like this is being used for ages by multiple companies, for example for DDOS protection and detection of bots .

**Comment by creestee** [comment]
But why did changing the user agent work?

**Comment by elmissouri16** [comment]
changing user-agent, delete session file all are temp solutions after some time u will face the same problem the weird thing that's I discover today that even with all this some accounts are still working fine and don't get any error 400 or anything, this is with a  old version of this API

**Comment by heldausberlin** [comment]
Changing User Agent won’t work, it is not the IP. Both changes didn’t work.
And I don’t think it is mouse movement detection. So it works for a while and stops immediately. 

I now disabled comments and it works since then. Maybe repetitive comments are the problem indeed as @svinc said.

So you all should try out first. And don’t forget to delete the session file. It won’t work once you were blocked.

**Comment by frankpantone** [comment]
I don't think it has anything to do with comments, I've disabled comments since day 1 and I still received 400 error on liking pictures, but as soon as I try to like a picture through the app, it works no problem. So I guess I just need to delete the session file and I'll be back up and running?

**Comment by mrallanodreman** [comment]
@frankpantone yes and now you have to forget the several actions for a time

Just adjust your liķes to 200 per day and go growing to take the 1200 or something like that! 

I suggest dont take several actions for a wile...

**Comment by frankpantone** [comment]
@mrallanodreman thanks for the heads up, I had my settings throttled down too before I got the 400 error. What did you mean to take the 1200?

**Comment by mrallanodreman** [comment]
@frankpantone you was using less than 400 actions per day?

**Comment by heldausberlin** [comment]
Since I disabled comments yesterday, the bot runs like clockwork again.

But it is important to delete/rename the [username].session file first.

**Comment by husligc** [comment]
> @svinc Idk, it sure looks like it . How does even an imacro running exactly the same follow commands I do manually get blocked and detected so easily ? The only thing I see missing is random mouse movement and clicks . I got blocked within 3 follows with an imacro just pressing follow button on a user list I have in a text file and then after logging out I repeated the same process manually and did 20 follows without issues .

Are we making any progress with this?  All of my bots are down with not liked 400.  It seems big changes are needed.

**Comment by elmissouri16** [comment]
just update setting the value of x-instagram-ajax  and X-ig-app-ID, randomize the delays seem promising  8 hours and its working fine with same session file and i didn't receive any spam errors

**Comment by mrallanodreman** [comment]
@elmissouri16 can you explain how to make this uodate?

**Comment by elmissouri16** [comment]
> 
> 
> @elmissouri16 can you explain how to make this update?

Im using an old version of this API lot of changes on it i made even before this problem of spam I well test more to be sure if this solves the problem  and i well share gist of the change i did

**Comment by danielcharrua** [comment]
Hi guys, I have updated to v0.5.7 with the new yml file and I have it running now for about 4 hours without 400 errors, yesterday I was using a previous version and after deleting session file I get blocked with the first like.. Are you using the latest version and using the yml config file?

**Comment by elmissouri16** [comment]
@danielcharrua  I'm using old one before the implementation of yml file and did some changes on it

**Comment by mrallanodreman** [comment]
Can you share it? Im usinh 1.2.6 version

**Comment by creestee** [comment]
I´m using 1.2.6 and now that I have changed the ua it works with comments since monday

**Comment by mrallanodreman** [comment]
@creestee meen you already explain how to change the user agent?

I want to add.or try with my android ua
Always i log in my bot, inatagram send me an email saying that are a log inusual...

I think when i change the ua to the user agent vinculated to my app/user the bot will camuflate... only if i use the same ua of my phone!

**Comment by frankpantone** [comment]
Is this consensus i'ts better to use an older version or new version?

**Comment by Pencilhead12** [comment]
@creestee Fantastic! Could you please explain how to change the user agent, and what you changed it to? :) Also, how would I go about getting version 1.2.6? And the last thing: does the example.py work also work for that version? ^_^

**Comment by danielcharrua** [comment]
I can confirm that after updating to the new version 0.5.7 the bot started working for one day, but today it has stopped working.. with the 400 error.. I will cool down for some days..

**Comment by mbe9477** [comment]
Hi @ all:
I tried hard to get the bot working again, but failed. Did I understand it right, that currently no stable version is available to get similar easy config done as before? 
Cheers

**Comment by creestee** [comment]
> After I changed the user agent it even works after restarting and without deleting the .session file!
> 
> To change it you have to edit instabot.py:
> Search for `self.user_agent =` and replace `random.sample(config.get("list_of_ua"), 1)[0]` with the ua of your device (you can look it up in the internet).

@Pencilhead12 @mrallanodreman

**Comment by danielcharrua** [comment]
@creestee your solution is working? Isn't a chance that you get banned on your mobile device?

**Comment by mrallanodreman** [comment]
Hey the solutiom of @creestee is working, change the user agent for the useragent for the app of your device mobile... 


"Instagram useragent instagram samsung a30"

**Comment by danielcharrua** [comment]
@mrallanodreman is working for more than 48 hours?

**Comment by mrallanodreman** [comment]
That will depends on you!
Im giving 100likes per day because i want to have a cooldown actions

**Comment by husligc** [comment]
> Hey the solutiom of @creestee is working, change the user agent for the useragent for the app of your device mobile...
> 
> "Instagram useragent instagram samsung a30"

Still not liked 400 even with very low settings. (140 comments, 160 likes) changed user agent to the same as my phone.

We need to drastically change something with the bot to have it perform anywhere close to like it was before this update. @svinc @dimisdas

**Comment by mrallanodreman** [comment]
If you got a account with 404 error you have to cooldown for st least 2 or 3 days before run the bot i  a low settings

**Comment by mrallanodreman** [comment]
@creestee your idea works fine for me

**Comment by alessandro893** [comment]
> UPDATE: There is a way to spoof UA as IG built in browser. Successfully logged in.
> Put "https://wtools.io/check-my-user-agent" to the "website" blank;
> Open the link in IG;
> then copy the UA and apply it to the bot;
> That's it :)
> ** It worked with the old version (python 2.7) bot by modifying ./src/userinfo.py by adding :
> **user_agent = ("the_ua_shown_above" "")**
> 
> 01.08.2019_10:40 Removing already liked medias..
> 01.08.2019_10:40 Trying to like media: 2100927239763767097
> 01.08.2019_10:40 Liked: 2100927239763767097. Like #1.
> 01.08.2019_10:42 Trying to like media: 2100926144195700601
> 01.08.2019_10:42 Liked: 2100926144195700601. Like #2.
> 01.08.2019_10:43 Trying to like media: 2100916801098352991
> 01.08.2019_10:43 Liked: 2100916801098352991. Like #3.
> 
> Request URL: https://www.instagram.com/web/likes/the_post_code/like/?hl=en
> Request Method: POST
> Status Code: 200

Thanks bro!
It started working after I changed UA in userinfo.py and deleted .session file.

**Comment by joaocpereiraa** [comment]
User agent methodo doesnt work for me.

**Comment by etaxi341** [comment]
Does not work for me too... crftoken also does not work.

**Comment by joaocpereiraa** [comment]
We need to wait for resolutions xb

**Comment by wdstk** [comment]
As of the current situation, I think the bot must behave like Instagram app and chances are;

1. Imitate HTTP Header and request perfectly
2. Encrypt requests like when using private API
[Reverse Engineering the Android Instagram App's Private API](http://eliasbagley.github.io/reverseengineering/2016/12/02/reverse-engineering-instagram-api.html)
3. Simulate human behavior; such as refreshing home, refreshing feed, accessing to suggested posts...

**Comment by neospeed83** [comment]
@wdstk sounds promising. I hope it does not take too long to go live though 😅

**Comment by joaocpereiraa** [comment]
any news? xb

**Comment by anzora95** [comment]
the instagram API is also compromised

**Comment by dsmh** [comment]
Hi everyone, does anybody have this bot working, I have been blocked with just 6 likes. I want to know how to change the user agent since I installed the software with pip3. 

Thanks in advance.

**Comment by spud-dog** [comment]
I have the same 'action blocked' issue after installing latest version. Switched bot off and left it for 24hrs and back on again, still blocked. 3 weeks ago it was doing 1800 likes and 650 comments a day. Pls find a solution.

**Comment by alessandro893** [comment]
> Hi everyone, does anybody have this bot working, I have been blocked with just 6 likes. I want to know how to change the user agent since I installed the software with pip3.
> 
> Thanks in advance.

You must edit userinfo.py file for changing user_agent (my is in "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/instabot_py/userinfo.py")

finally i got it working (instabot 0.5.9):

I wrote a script to restarting instabot:

pid=$(ps -ax | grep instabot | grep py | awk '{ print $1 }')
kill="/bin/kill"

login="--login yourlogin"
pass="--password yourpass"
config="-c /opt/local/etc/instabot/instabot.config.yml"
session="/opt/local/etc/instabot/yourlogin.session"

instabot="/opt/local/Library/Frameworks/Python.framework/Versions/3.7/bin/instabot-py"
python="/opt/local/bin/python3.7"


if (test "$1" = "--stop")
then
  if (test "$pid" > "0")
  then
    $kill $pid
    echo pid "$pid" - stopped
  else
    echo no pid, already stopped
  fi
fi

if (test "$1" = "--start")
then
  if (test "$pid" > "0")
  then
    echo pid "$pid" - already started
  else
    echo starting
    $python $instabot $login $pass $config > "/private/var/Log/instabot/$(date +%dday-\(%H\:%M\)).log"
  fi
fi

if (test "$1" = "--restart")
then
  $kill $pid
  echo restarting
  $python $instabot $login $pass $config > "/private/var/Log/instabot/$(date +%dday-\(%H\:%M\)).log"
fi

if (test "$1" = "--newstart")
then
  $kill $pid
  rm $session
  echo Session file removed
  echo starting
  $python $instabot $login $pass $config > "/private/var/Log/instabot/$(date +%dday-\(%H\:%M\)).log"
fi


(in instabot.py I added this at the top:)
import shlex, subprocess

(and after string "self.persistence.insert_media(media_id=media_id, status="400", )" 
I added this:) 
subprocess.run('/pathToScript/instabot --newstart', shell=True)

**Comment by dsmh** [comment]
> pid=$(ps -ax | grep instabot | grep py | awk '{ print $1 }')
> kill="/bin/kill"
> 
> login="--login yourlogin"
> pass="--password yourpass"
> config="-c /opt/local/etc/instabot/instabot.config.yml"
> session="/opt/local/etc/instabot/yourlogin.session"
> 
> instabot="/opt/local/Library/Frameworks/Python.framework/Versions/3.7/bin/instabot-py"
> python="/opt/local/bin/python3.7"
> 
> if (test "$1" = "--stop")
> then
> if (test "$pid" > "0")
> then
> $kill $pid
> echo pid "$pid" - stopped
> else
> echo no pid, already stopped
> fi
> fi
> 
> if (test "$1" = "--start")
> then
> if (test "$pid" > "0")
> then
> echo pid "$pid" - already started
> else
> echo starting
> $python $instabot $login $pass $config > "/private/var/Log/instabot/$(date +%dday-(%H:%M)).log"
> fi
> fi
> 
> if (test "$1" = "--restart")
> then
> $kill $pid
> echo restarting
> $python $instabot $login $pass $config > "/private/var/Log/instabot/$(date +%dday-(%H:%M)).log"
> fi
> 
> if (test "$1" = "--newstart")
> then
> $kill $pid
> rm $session
> echo Session file removed
> echo starting
> $python $instabot $login $pass $config > "/private/var/Log/instabot/$(date +%dday-(%H:%M)).log"
> fi

Thank you so much for the information, I added the user agent on the file, however, I'm unable to run your script for restanting instabot-py, since I'm running 7 accounts on my raspi and this script seem to be written for just one account. However I'm running a simple script to delete all the .session information at the end of the bot, just missing the killing when the error 400 appears.

I'm having some luck whit the change of the user agent so far... will keep testing.

**Comment by creestee** [comment]
changed user agent now works over two weeks for me

**Comment by mrallanodreman** [comment]
@creestee confirm

**Comment by creestee** [comment]
yesterday it sadly stopped working. I now have deleted the session file again and it´s runnung for now.

**Comment by asu01928** [comment]
Get Media by tag: follow4follow
2019-10-15 04:03:41,826 - InstaBot - DEBUG - Trying to like media: 2154967220661125496, https://www.instagram.com/p/B3n-0UKHgF4/
2019-10-15 04:03:41,950 - InstaBot - INFO - Not liked: 400 message {"message": "feedback_required", "spam": true, "feedback_title": "Action Blocked", "feedback_message": "This action was blocked. Please try again later. We restrict certain content and actions to protect our community. Tell us if you think we made a mistake.", "feedback_url": "repute/report_problem/instagram_like_add/", "status": "fail"}work for 6hours and 300 account in my vps got message like that XD but in my account i can like or vomment

**Comment by mrallanodreman** [comment]
> 
> 
> yesterday it sadly stopped working. I now have deleted the session file again and it´s runnung for now.

instagram is detecting the activity from windows, even when i change the user agent...

**Comment by nickgryg** [comment]
This is very outdated issue. If you still have it, please try to use instabot-py version 0.5.20 or later. You can set custom User Agent starting from version 0.5.15.
